### DISCUSSION TITLE
Question about Chip select Pin

### DISCUSSION THREAD

**Discussion by carl336** [micropython/micropython#14327] [RP2040 / Pico]
Hello, I am a beginner, learning Micropython with Raspberry Pico.

I'm using the PIO mode to stimulate SPI signal.

First of all, here is the code that I wrote:
https://paste.fastmirror.net/?31f1116d1dfba6df#2B31qwCJhPRq3Av8h9VXxgP7bzKYS69FMC5PUVhNYVpQ

Here is the diagram that tested by a logic analyzer:
https://imgur.com/cCTkiSQ

And here is the way how I connected Pico with logic analyzer:

GP10       SCK　                 →Channel 0
GP12       MOSI                  →Channel 1
GP11       SCS                    →Channel 2
GP 17       DISP                   →Channel 3
GP 19       EXTCOMIN         →Channel 4
GP 2         VDD                    →Channel 5

Here are a specification of a LED Driver that I will try to study and try to turn on a back light later:
https://fastupload.io/ICQxJmtii5ffxRf/file

Here is the LCD that I would like to control:
https://fastupload.io/qy4K7XZTrNYBhrl/file
Module number: LS022B7DH03

On page 14, it says that the SIN can be set to logic HIGH, while the CLK_I is required to be always active and continuous, as CLK_I provides the reference clock for internal circuits.

At the moment, what I want to do is:
Combine 0x20 and 0x00 together and send them together when SCS is high. Rather than separating 0x20 and 0x00 as shown in the original timing diagram. I want to send 16 bits at once instead of splitting them into two 8-bit transmissions.

https://imgur.com/wFKX6KF

Can anybody tell me how to do it?

**Comment by peterhinch** [comment]
I don't have a detailed answer to your question but there is some background information on [sharp displays](https://github.com/peterhinch/micropython-nano-gui/blob/master/DRIVERS.md#4-drivers-for-sharp-displays) for a device driver that I wrote for Adafruit modules. They do have some unusual requirements, and the driver must take steps not to polarise the display.

My honest opinion of these displays is that they have limited application. Their visibility is not great indoors - better outside. For most purposes ePaper has much better contrast, albeit with long update times. The Sharp displays are fast.
